inst/templates/ad_advs.R

```diff
@@ -0,0 +1,552 @@
+# (https://pharmaverse.github.io/admiral/articles/visits_periods.html#visits)
+advs <- advs %>%
+  # Derive Timing
+  mutate(
+    ATPTN = VSTPTNUM,
+    ATPT = VSTPT,
+    AVISIT = case_when(
+      str_detect(VISIT, "UNSCHED|RETRIEVAL|AMBUL") ~ NA_character_,
+      !is.na(VISIT) ~ str_to_title(VISIT),
+      TRUE ~ NA_character_
+    ),
+    AVISITN = as.numeric(case_when(
+      VISIT == "SCREENING 1" ~ "-1",
+      VISIT == "BASELINE" ~ "0",
+      str_detect(VISIT, "WEEK") ~ str_trim(str_replace(VISIT, "WEEK", "")),
+      TRUE ~ NA_character_
+    ))
+  )
+
+# Add Current HEIGHT Temporary variable (in cm)
+## Calculate Current HEIGHT/LENGTH at each time point Temporary variable----
+hgttmp <- advs %>%
+  filter((PARAMCD == "HEIGHT" | PARAMCD == "LENGTH") & VSSTRESU == "cm") %>%
+  select(STUDYID, USUBJID, HGTTMP = AVAL, HGTTMPU = VSSTRESU, AVISIT)
+
+advs <- advs %>%
+  derive_vars_merged(
+    dataset_add = hgttmp,
+    by_vars = exprs(STUDYID, USUBJID, AVISIT)
+  )
+
+## Merge ADVS to the chosen Growth metadata as an input to meta_criteria ----
+## Calculate z-scores/percentiles
+## Calculate Weight for AGE z-score and Percentile ----
+## Note: PARAMN needs to be updated.
+advs_age <- advs %>% derive_params_growth_age(
+  sex = SEX,
+  age = AAGECUR,
+  age_unit = AAGECURU,
+  meta_criteria = weight_for_age,
+  parameter = VSTESTCD == "WEIGHT",
+  analysis_var = AVAL,
+  set_values_to_sds = exprs(
+    PARAMCD = "WTASDS",
+    PARAM = "Weight-for-age z-score"
+  ),
+  set_values_to_pctl = exprs(
+    PARAMCD = "WTAPCTL",
+    PARAM = "Weight-for-age percentile"
+  )
+)
+
+## Calculate BMI for AGE z-score and Percentile ----
+## Note: PARAMN needs to be updated for z-score and percentile in final dataset.
+advs_age <- advs_age %>% derive_params_growth_age(
+  sex = SEX,
+  age = AAGECUR,
+  age_unit = AAGECURU,
+  meta_criteria = bmi_for_age,
+  parameter = VSTESTCD == "BMI",
+  analysis_var = AVAL,
+  set_values_to_sds = exprs(
+    PARAMCD = "BMISDS",
+    PARAM = "BMI-for-age z-score"
+  ),
+  set_values_to_pctl = exprs(
+    PARAMCD = "BMIPCTL",
+    PARAM = "BMI-for-age percentile"
+  )
+)
+
+## Calculate Head Circumference for AGE z-score and Percentile ----
+## Note: PARAMN needs to be updated for z-score and percentile in final dataset.
+advs_age <- advs_age %>% derive_params_growth_age(
+  sex = SEX,
+  age = AAGECUR,
+  age_unit = AAGECURU,
+  meta_criteria = who_hc_for_age,
+  parameter = VSTESTCD == "HDCIRC",
+  analysis_var = AVAL,
+  set_values_to_sds = exprs(
+    PARAMCD = "HDCSDS",
+    PARAM = "HDC-for-age z-score"
+  ),
+  set_values_to_pctl = exprs(
+    PARAMCD = "HDCPCTL",
+    PARAM = "HDC-for-age percentile"
+  )
+)
+
+## Calculate Z-Score and percentile parameters for Height/Length ----
+message("To derive height/length parameters, below function needs to call separately
+        for Height and Length based on the input data and current age of the patient,
+        as it depends on your CRF guidelines.")
```

Comment:
@rossfarrugia We can update ADVS peds template once `derive_params_growth_height()` is updated accordingly.

Reply:
@rossfarrugia We can keep it as is, as it was discussed to make separate calls for HEIGHT and LENGTH.

Reply:
agree - only i would then restructure a bit, e.g. currently you combine all this metadata as one and then split it out again via `who_wt_for_ht_lgth %>% filter(MEASURE == "LENGTH")`. this doesn't make sense to me as you could just make 1 `who_wt_for_ht` and 1 `who_wt_for_lgth` instead at the top metadata section. and you dont even need a `HEIGHT_LENGTH` variable there anymore for example.

Reply:
you mean the template draft outline doc on our MS teams? thats just a starter to get us building these issues/PRs. your PR is far more advanced than that now so i've not been maintaining that working word doc, e.g. it still talks of doing these parameters by one function call whereas Zelos found better to run the function twice. we can leave that doc now and instead focus on the latest in our PRs as the latest plan. we don't really maintain "specs" in same way as you do when stats programming here, its mainly just the GH issue and then the PR review discussoins that guide us

Reply:
and ultimatelty the vignette will then explain everything to users in the end

Reply:
sorry @Lina2689 @Fanny-Gautier at the team call today i realised i had incorrectly said above "you dont even need a HEIGHT_LENGTH variable" - i meant `MEASURE` here as we don't need any variable to say "Height" or "Length" if we no longer join these into one metadata file. We still should make the values and units consistent so that they work with the function from @zdz2101 (i.e. `HEIGHT_LENGTH` and `HEIGHT_LENGTHU`)